bigtable/result_row.go

```diff
@@ -0,0 +1,801 @@
+				if err := assignValue(destKey, srcKey.Interface()); err != nil {
+					return fmt.Errorf("error assigning map key type %s to %s: %w", srcKeyType, destKeyType, err)
+				}
+				if err := assignValue(destValue, srcValElem.Interface()); err != nil {
+					return fmt.Errorf("error assigning map value type %s to %s: %w", srcValElem.Type(), destValType, err)
+				}
+				dest.SetMapIndex(destKey, destValue)
+			}
+			return nil
+		}
+
+		// Case: Assigning map[K]V source to map[K]*V destination (Allocate pointers)
+		if destKeyType == srcKeyType && destValType.Kind() == reflect.Ptr && srcValType == destValType.Elem() {
+			if dest.IsNil() {
+				dest.Set(reflect.MakeMap(destType))
+			}
+			mapIter := srcVal.MapRange()
+			for mapIter.Next() {
+				srcKey := mapIter.Key()
+				srcValue := mapIter.Value() // K and V
+				// Need new instances for map SetMapIndex
+				destKey := reflect.New(destKeyType).Elem()
+				destValPtr := reflect.New(destValType).Elem() // Destination element *V
+				// Assign K to K
+				if err := assignValue(destKey, srcKey.Interface()); err != nil {
+					return fmt.Errorf("error assigning map key type %s to %s: %w", srcKeyType, destKeyType, err)
+				}
+				// Assign V to *V (will allocate pointer)
+				if err := assignValue(destValPtr, srcValue.Interface()); err != nil {
+					return fmt.Errorf("error assigning map value type %s to %s: %w", srcValType, destValType, err)
+				}
+				dest.SetMapIndex(destKey, destValPtr)
+			}
+			return nil
+		}
+
+		// Case: Assigning map[K]V or map[K]*V source to map[string]any
+		// If dest is map[string]any, destValType will be anyType (interface{})
+		if destKeyType.Kind() == reflect.String && destValType.Kind() == reflect.Interface {
+			// Source map keys must be assignable/convertible to string
+			// Check if keys are compatible string types
+			if !srcKeyType.AssignableTo(destKeyType) && !(srcKeyType.Kind() == reflect.String && destKeyType.Kind() == reflect.String) {
+				// If srcKeyType is []byte, do not allow conversion
+				return fmt.Errorf("cannot assign source map with key type %s to destination map with key type %s", srcKeyType, destKeyType)
+			}
+			if dest.IsNil() {
+				dest.Set(reflect.MakeMap(destType))
+			}
+			mapIter := srcVal.MapRange()
+			for mapIter.Next() {
+				srcKey := mapIter.Key()     // K
+				srcValue := mapIter.Value() // V or *V
+				// Key: Assume string assignable or kind string
+				destKey := reflect.ValueOf(srcKey.Convert(destKeyType).Interface())
+				// Value: Assign V or *V to interface{} element
+				destValue := reflect.New(destValType).Elem()
+				if err := assignValue(destValue, srcValue.Interface()); err != nil {
+					return fmt.Errorf("error assigning map value type %s to interface{}: %w", srcValue.Type(), err)
+				}
+				dest.SetMapIndex(destKey, destValue)
+			}
+			return nil
+		}
+	}
+
+	// Common numeric conversions (allow assigning int64 to int, float64 to float32 etc.)
+	if srcVal.CanConvert(dest.Type()) {
```

Comment:
I would vote against allowing implicit conversions like this. I think we should make users pass the exact types and they can convert the output as they wish.